### PR TITLE
[4.0] Correcting module xtd wrong js

### PR DESCRIPTION
**Pull request by infograf768**
### Summary of Changes 
The `admin-modules-modal.es6.js` used for the module xtd was wrong as it was containing the code present in cpanel `admin-add_module.es6.js`

I only pasted the code from J3 here in a new `admin-modules-modal.js` as I fail to make it an es6.

Need help for that.
@Fedik @dgrammatiko 

In the mean while, can be tested.

[EDIT] The js is now es6

### Testing Instructions
Patch, run npm and then edit an article. Insert a module through the CMS Content dropdown in TinyMCE

Screen Shot 2020-01-12 at 16 17 34

### Before patch
Clicking on a module in the modal does not work

### After patch
Works fine

### CODE REVIEW

**Comment by dgrammatiko** [comment]
There you go:
```js
/**
  * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */

document.addEventListener('DOMContentLoaded', () => {
    "use strict";

    /** Get the elements **/
    const modulesLinks = [].slice.call(document.querySelectorAll('.js-module-insert'));
    const positionsLinks = [].slice.call(document.querySelectorAll('.js-position-insert'));

    /** Assign listener for click event (for single module id insertion) **/
    modulesLinks.forEach((element) => {
        element.addEventListener('click', (event) => {
            event.preventDefault();
            const modid = event.target.getAttribute('data-module');
            const editor = event.target.getAttribute('data-editor');

            /** Use the API, if editor supports it **/
            if (window.parent.Joomla && window.parent.Joomla.editors && window.parent.Joomla.editors.instances && window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {
                window.parent.Joomla.editors.instances[editor].replaceSelection("{loadmoduleid " + modid + "}")
            } else {
                window.parent.jInsertEditorText("{loadmoduleid " + modid + "}", editor);
            }
        });
    });

    /** Assign listener for click event (for position insertion) **/
    positionsLinks.forEach((element) => {
        element.addEventListener('click', function (event) {
            event.preventDefault();
            const position = event.target.getAttribute('data-position');
            const editor = event.target.getAttribute('data-editor');

            /** Use the API, if editor supports it **/
            if (window.Joomla && window.Joomla.editors && Joomla.editors.instances && Joomla.editors.instances.hasOwnProperty(editor)) {
                Joomla.editors.instances[editor].replaceSelection("{loadposition " + position + "}")
            } else {
                window.parent.jInsertEditorText("{loadposition " + position + "}", editor);
            }
        });
    });
});
```

**Comment by dgrammatiko** [comment]
Actually the else part is useless in J4 so:
```js
/**
  * @copyright  Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
  * @license    GNU General Public License version 2 or later; see LICENSE.txt
  */

document.addEventListener('DOMContentLoaded', () => {
    "use strict";

    /** Get the elements **/
    const modulesLinks = [].slice.call(document.querySelectorAll('.js-module-insert'));
    const positionsLinks = [].slice.call(document.querySelectorAll('.js-position-insert'));

    /** Assign listener for click event (for single module id insertion) **/
    modulesLinks.forEach((element) => {
        element.addEventListener('click', (event) => {
            event.preventDefault();
            const modid = event.target.getAttribute('data-module');
            const editor = event.target.getAttribute('data-editor');

            /** Use the API **/
            if (window.parent.Joomla && window.parent.Joomla.editors && window.parent.Joomla.editors.instances && window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {
                window.parent.Joomla.editors.instances[editor].replaceSelection("{loadmoduleid " + modid + "}")
            }
        });
    });

    /** Assign listener for click event (for position insertion) **/
    positionsLinks.forEach((element) => {
        element.addEventListener('click', function (event) {
            event.preventDefault();
            const position = event.target.getAttribute('data-position');
            const editor = event.target.getAttribute('data-editor');

            /** Use the API **/
            if (window.Joomla && window.Joomla.editors && Joomla.editors.instances && Joomla.editors.instances.hasOwnProperty(editor)) {
                Joomla.editors.instances[editor].replaceSelection("{loadposition " + position + "}")
            }
        });
    });
});
```

**Comment by dgrammatiko** [comment]
@infograf768 my first commit build/media_src/com_modules/js/admin-modules-modal.es6.js has also some code for closing the modal, I guess you should revert to that and ignore my above comments. BTW I have no clue why they rewrote it with the new design of the template, probably you should check the dashboard after changing this file (?)

**Comment by infograf768** [comment]
@dgrammatiko 
Close button works fine with your second proposal.
> BTW I have no clue why they rewrote it with the new design of the template, probably you should check the dashboard after changing this file (?)

I had checked already. It's fine.

remains to solve Hound...

**Comment by dgrammatiko** [comment]
tabs to spaces and line length, should be easy

**Comment by infograf768** [comment]
@dgrammatiko 
yes for spaces and tabs but what about
`Do not access Object.prototype method 'hasOwnProperty' from target object no-prototype-builtins` ?

**Comment by dgrammatiko** [comment]
https://eslint.org/docs/rules/no-prototype-builtins
or 
```js
Object.prototype.hasOwnProperty.call(window.parent.Joomla.editors.instances, editor)
```

PS sorry my PC is really broken at the moment

**Comment by infograf768** [comment]
@dgrammatiko 
I think I solved all except
`Do not access Object.prototype method 'hasOwnProperty' from target object no-prototype-builtins

in line 23
`        && window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {`

I did not understand your comment.

File: build/media_source/com_modules/js/admin-modules-modal.es6.js [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
+  modulesLinks.forEach((element) => {
+    element.addEventListener('click', (event) => {
+      event.preventDefault();
+      const modid = event.target.getAttribute('data-module');
+      const editor = event.target.getAttribute('data-editor');
+
+      // Use the API
+      if (window.parent.Joomla && window.parent.Joomla.editors
+        && window.parent.Joomla.editors.instances
+        && window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {
```
[the remark this answers is not in the document]
Comment (dgrammatiko) [reply build/media_source/com_modules/js/admin-modules-modal.es6.js]: @infograf768 replace
```js
&& window.parent.Joomla.editors.instances.hasOwnProperty(editor)) {
```
with 
```js
&& Object.prototype.hasOwnProperty.call(window.parent.Joomla.editors.instances, editor)) {
```

**Comment by infograf768** [comment]
@dgrammatiko 
remains 
`Unexpected string concatenation prefer-template`
I had tried
`Joomla.editors.instances[editor].replaceSelection('{loadposition ${position}}');
but it also throwed an error.

File: build/media_source/com_modules/js/admin-modules-modal.es6.js [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
+    element.addEventListener('click', (event) => {
+      event.preventDefault();
+      const modid = event.target.getAttribute('data-module');
+      const editor = event.target.getAttribute('data-editor');
+
+      // Use the API
+      if (window.parent.Joomla && window.parent.Joomla.editors
+        && window.parent.Joomla.editors.instances
+        && Object.prototype.hasOwnProperty.call(window.parent.Joomla.editors.instances, editor)) {
+        window.parent.Joomla.editors.instances[editor].replaceSelection('{loadmoduleid ' + modid + '}');
```
[the remark this answers is not in the document]
Comment (dgrammatiko) [reply build/media_source/com_modules/js/admin-modules-modal.es6.js]: Replace
```js
'{loadmoduleid ' + modid + '}'
```
with
```js
`{loadmoduleid ${modid}}`
```

**Comment by infograf768** [comment]
Maybe just
`// eslint-disable-line prefer-template`

what you think?

**Comment by dgrammatiko** [comment]
It needs to be backtick not single quote!

**Comment by infograf768** [comment]
Ok, will do tomorrow

**Comment by infograf768** [comment]
@dgrammatiko done
@wilsonge 
Hound is still throwing a violation, but no specific message.
Can you help?

**Comment by infograf768** [comment]
All should be OK now.
Please test.

**Comment by jwaisner** [comment]
Modal now selects module after testing. Comparing to J3 the modal should close after selecting the module but does not in this PR.

**Comment by infograf768** [comment]
Auto-close done.
Inserting module position done

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on e7ef6f5c4aeec819e333a57d7993ed91dfb5dc74<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27499">issues.joomla.org/tracker/joomla-cms/27499</a>.</sub>

**Comment by N6REJ** [comment]
I have tested this item :white_check_mark: successfully on e7ef6f5c4aeec819e333a57d7993ed91dfb5dc74<br /><br />installed this mornings build of J4, installed patch via patch tester, ran npm ci, refreshed article, attempted to insert module & nothing happened.  Tried article and it was fine.

reverted patch, reapplied patch, did NOT run npm again, and it worked fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27499">issues.joomla.org/tracker/joomla-cms/27499</a>.</sub>

**Comment by infograf768** [comment]
rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27499">issues.joomla.org/tracker/joomla-cms/27499</a>.</sub>

**Comment by HLeithner** [comment]
Thanks

**Pull request merged**